// ng2-bs3-modal/bundles/ng2-bs3-modal.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
System.register("ng2-bs3-modal/components/modal", ['angular2/core'], function(exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var core_1;
    var ModalComponent, ModalSize, ModalResult, id;
    function uniqueId(prefix) {
        return prefix + ++id;
    }
    exports_1("uniqueId", uniqueId);
    return {
        setters:[
            function (core_1_1) {
                core_1 = core_1_1;
            }],
        execute: function() {
            ModalComponent = (function () {
                function ModalComponent() {
                    this.id = uniqueId('modal_');
                    this.result = ModalResult.None;
                    this.hiding = false;
                    this.overrideSize = null;
                    this.animation = true;
                    this.onClose = new core_1.EventEmitter();
                }
                ModalComponent.prototype.init = function () {
                    var _this = this;
                    this.$modal = jQuery('#' + this.id);
                    this.$modal.appendTo('body').modal({ show: false });
                    this.$modal
                        .off('hide.bs.modal.ng2-bs3-modal')
                        .on('hide.bs.modal.ng2-bs3-modal', function (e) {
                        _this.hiding = true;
                        if (_this.result === ModalResult.None)
                            _this.dismiss();
                        _this.result = ModalResult.None;
                    })
                        .off('hidden.bs.modal.ng2-bs3-modal')
                        .on('hidden.bs.modal.ng2-bs3-modal', function (e) {
                        _this.hiding = false;
                        _this.overrideSize = null;
                    });
                };
                ModalComponent.prototype.ngOnDestroy = function () {
                    if (this.$modal) {
                        this.$modal.data('bs.modal', null);
                        this.$modal.remove();
                    }
                };
                ModalComponent.prototype.open = function (size) {
                    this.init();
                    if (ModalSize.validSize(size))
                        this.overrideSize = size;
                    this.$modal.modal('show');
                };
                ModalComponent.prototype.close = function () {
                    this.result = ModalResult.Close;
                    this.onClose.next(this.result);
                    this.hide();
                };
                ModalComponent.prototype.dismiss = function () {
                    this.result = ModalResult.Dismiss;
                    this.onClose.next(this.result);
                    this.hide();
                };
                ModalComponent.prototype.hide = function () {
                    if (!this.hiding)
                        this.$modal.modal('hide');
                };
                ModalComponent.prototype.isSmall = function () {
                    return this.overrideSize !== ModalSize.Large && this.size === ModalSize.Small || this.overrideSize === ModalSize.Small;
                };
                ModalComponent.prototype.isLarge = function () {
                    return this.overrideSize !== ModalSize.Small && this.size === ModalSize.Large || this.overrideSize === ModalSize.Large;
                };
                __decorate([
                    core_1.Input(), 
                    __metadata('design:type', Boolean)
                ], ModalComponent.prototype, "animation", void 0);
                __decorate([
                    core_1.Input(), 
                    __metadata('design:type', String)
                ], ModalComponent.prototype, "backdrop", void 0);
                __decorate([
                    core_1.Input(), 
                    __metadata('design:type', Boolean)
                ], ModalComponent.prototype, "keyboard", void 0);
                __decorate([
                    core_1.Input(), 
                    __metadata('design:type', String)
                ], ModalComponent.prototype, "size", void 0);
                __decorate([
                    core_1.Output(), 
                    __metadata('design:type', core_1.EventEmitter)
                ], ModalComponent.prototype, "onClose", void 0);
                ModalComponent = __decorate([
                    core_1.Component({
                        selector: 'modal',
                        template: "\n        <div id=\"{{id}}\" class=\"modal\" [ngClass]=\"{ fade: animation }\" tabindex=\"-1\" role=\"dialog\"\n            [attr.data-keyboard]=\"keyboard\" [attr.data-backdrop]=\"backdrop\">\n            <div class=\"modal-dialog\" [ngClass]=\"{ 'modal-sm': isSmall(), 'modal-lg': isLarge() }\">\n                <div class=\"modal-content\">\n                    <ng-content></ng-content>\n                </div>\n            </div>\n        </div>\n    "
                    }), 
                    __metadata('design:paramtypes', [])
                ], ModalComponent);
                return ModalComponent;
            }());
            exports_1("ModalComponent", ModalComponent);
            ModalSize = (function () {
                function ModalSize() {
                }
                ModalSize.validSize = function (size) {
                    return size && (size === ModalSize.Small || size === ModalSize.Large);
                };
                ModalSize.Small = 'sm';
                ModalSize.Large = 'lg';
                return ModalSize;
            }());
            exports_1("ModalSize", ModalSize);
            (function (ModalResult) {
                ModalResult[ModalResult["None"] = 0] = "None";
                ModalResult[ModalResult["Close"] = 1] = "Close";
                ModalResult[ModalResult["Dismiss"] = 2] = "Dismiss";
            })(ModalResult || (ModalResult = {}));
            exports_1("ModalResult", ModalResult);
            id = 0;
        }
    }
});
System.register("ng2-bs3-modal/components/modal-header", ['angular2/core', "ng2-bs3-modal/components/modal"], function(exports_2, context_2) {
    "use strict";
    var __moduleName = context_2 && context_2.id;
    var core_2, modal_1;
    var ModalHeaderComponent;
    return {
        setters:[
            function (core_2_1) {
                core_2 = core_2_1;
            },
            function (modal_1_1) {
                modal_1 = modal_1_1;
            }],
        execute: function() {
            ModalHeaderComponent = (function () {
                function ModalHeaderComponent(modal) {
                    this.modal = modal;
                    this.showClose = false;
                }
                __decorate([
                    core_2.Input('show-close'), 
                    __metadata('design:type', Boolean)
                ], ModalHeaderComponent.prototype, "showClose", void 0);
                ModalHeaderComponent = __decorate([
                    core_2.Component({
                        selector: 'modal-header',
                        template: "\n        <div class=\"modal-header\">\n            <button type=\"button\" class=\"close\" data-dismiss=\"modal\" aria-label=\"Close\" (click)=\"modal.dismiss()\" [hidden]=\"!showClose\"><span aria-hidden=\"true\">&times;</span></button>\n            <ng-content></ng-content>\n        </div>\n    "
                    }), 
                    __metadata('design:paramtypes', [modal_1.ModalComponent])
                ], ModalHeaderComponent);
                return ModalHeaderComponent;
            }());
            exports_2("ModalHeaderComponent", ModalHeaderComponent);
        }
    }
});
System.register("ng2-bs3-modal/components/modal-body", ['angular2/core', "ng2-bs3-modal/components/modal"], function(exports_3, context_3) {
    "use strict";
    var __moduleName = context_3 && context_3.id;
    var core_3, modal_2;
    var ModalBodyComponent;
    return {
        setters:[
            function (core_3_1) {
                core_3 = core_3_1;
            },
            function (modal_2_1) {
                modal_2 = modal_2_1;
            }],
        execute: function() {
            ModalBodyComponent = (function () {
                function ModalBodyComponent(modal) {
                    this.modal = modal;
                }
                ModalBodyComponent = __decorate([
                    core_3.Component({
                        selector: 'modal-body',
                        template: "\n        <div class=\"modal-body\">\n            <ng-content></ng-content>\n        </div>\n    "
                    }), 
                    __metadata('design:paramtypes', [modal_2.ModalComponent])
                ], ModalBodyComponent);
                return ModalBodyComponent;
            }());
            exports_3("ModalBodyComponent", ModalBodyComponent);
        }
    }
});
System.register("ng2-bs3-modal/components/modal-footer", ['angular2/core', "ng2-bs3-modal/components/modal"], function(exports_4, context_4) {
    "use strict";
    var __moduleName = context_4 && context_4.id;
    var core_4, modal_3;
    var ModalFooterComponent;
    return {
        setters:[
            function (core_4_1) {
                core_4 = core_4_1;
            },
            function (modal_3_1) {
                modal_3 = modal_3_1;
            }],
        execute: function() {
            ModalFooterComponent = (function () {
                function ModalFooterComponent(modal) {
                    this.modal = modal;
                    this.showDefaultButtons = false;
                }
                __decorate([
                    core_4.Input('show-default-buttons'), 
                    __metadata('design:type', Boolean)
                ], ModalFooterComponent.prototype, "showDefaultButtons", void 0);
                ModalFooterComponent = __decorate([
                    core_4.Component({
                        selector: 'modal-footer',
                        styles: ["\n        .btn[hidden] { display: none; }\n    "],
                        template: "\n        <div class=\"modal-footer\">\n            <ng-content></ng-content>\n            <button type=\"button\" class=\"btn btn-default\" data-dismiss=\"modal\" (click)=\"modal.dismiss()\" [hidden]=\"!showDefaultButtons\">Close</button>\n            <button type=\"button\" class=\"btn btn-primary\" (click)=\"modal.close()\" [hidden]=\"!showDefaultButtons\">Save</button>\n        </div>\n    "
                    }), 
                    __metadata('design:paramtypes', [modal_3.ModalComponent])
                ], ModalFooterComponent);
                return ModalFooterComponent;
            }());
            exports_4("ModalFooterComponent", ModalFooterComponent);
        }
    }
});
System.register("ng2-bs3-modal/ng2-bs3-modal", ["ng2-bs3-modal/components/modal", "ng2-bs3-modal/components/modal-header", "ng2-bs3-modal/components/modal-body", "ng2-bs3-modal/components/modal-footer"], function(exports_5, context_5) {
    "use strict";
    var __moduleName = context_5 && context_5.id;
    var modal_4, modal_header_1, modal_body_1, modal_footer_1;
    var MODAL_DIRECTIVES;
    var exportedNames_1 = {
        'MODAL_DIRECTIVES': true
    };
    function exportStar_1(m) {
        var exports = {};
        for(var n in m) {
            if (n !== "default"&& !exportedNames_1.hasOwnProperty(n)) exports[n] = m[n];
        }
        exports_5(exports);
    }
    return {
        setters:[
            function (modal_4_1) {
                modal_4 = modal_4_1;
                exportStar_1(modal_4_1);
            },
            function (modal_header_1_1) {
                modal_header_1 = modal_header_1_1;
                exportStar_1(modal_header_1_1);
            },
            function (modal_body_1_1) {
                modal_body_1 = modal_body_1_1;
                exportStar_1(modal_body_1_1);
            },
            function (modal_footer_1_1) {
                modal_footer_1 = modal_footer_1_1;
                exportStar_1(modal_footer_1_1);
            }],
        execute: function() {
            exports_5("MODAL_DIRECTIVES", MODAL_DIRECTIVES = [
                modal_4.ModalComponent,
                modal_header_1.ModalHeaderComponent,
                modal_body_1.ModalBodyComponent,
                modal_footer_1.ModalFooterComponent
            ]);
        }
    }
});
